Softeq.XToolkit.WhiteLabel.Droid/ActivityBase.cs

```diff
@@ -66,21 +74,32 @@ protected void AddViewForViewModel(ViewModelBase viewModel, int containerId)
     public abstract class ActivityBase<TViewModel> : ActivityBase
         where TViewModel : ViewModelBase
     {
-        protected TViewModel _viewModel;
-        protected List<Binding> Bindings { get; } = new List<Binding>();
+        private const string ShouldRestoreStateKey = "shouldRestore";
+        private readonly Lazy<IJsonSerializer> _jsonSerializer;
+        protected List<Binding> Bindings { get; }
+        protected virtual TViewModel ViewModel { get; set; }
 
-        public virtual TViewModel ViewModel => _viewModel ?? (_viewModel = ServiceLocator.Resolve<TViewModel>());
+        protected ActivityBase()
+        {
+            Bindings = new List<Binding>();
+            _jsonSerializer = new Lazy<IJsonSerializer>(ServiceLocator.Resolve<IJsonSerializer>);
+        }
 
         protected override void OnCreate(Bundle savedInstanceState)
         {
             base.OnCreate(null);
 
-            RequestedOrientation = ScreenOrientation.Portrait;
+            if (ViewModel == null)
+            {
+                ViewModel = (TViewModel)PageNavigation.Value.GetExistingOrCreateViewModel(typeof(TViewModel));
```

Comment:
I think we should move GetExistingOrCreateViewModel method out of PageNavigation class

Reply:
create seapate service